test/index.js

```diff
@@ -12,6 +12,14 @@ describe('union type', function() {
     var Point = Type({Point: [isNumber, isNumber]});
     assert.equal('function', typeof Point.Point);
   });
+  it('type can be used with instanceof', function() {
+    var Point = Type({Point: [isNumber, isNumber]});
+    assert.equal(Point.Point(4,2) instanceof Point, true);
+  });
+  it('constructor can be used with instanceof', function() {
+    var Point = Type({Point: [isNumber, isNumber]});
+    assert.equal(Point.Point(4,2) instanceof Point.Point, true);
```

Comment:
I agree!

Instead, we should create one predicate per data constructor. For example:

```javascript
//    Maybe :: Type
const Maybe = Type({Nothing: [], Just: [R.T]});

//    Maybe.isNothing :: Maybe a -> Boolean
//    Maybe.isJust :: Maybe a -> Boolean

Maybe.isNothing(Maybe.Nothing());   // => true
Maybe.isNothing(Maybe.Just(42));    // => false
Maybe.isJust(Maybe.Nothing());      // => false
Maybe.isJust(Maybe.Just(42));       // => true
```